### PR TITLE
[Improve][Connector-V2] Improve fake source connector

### PR DESCRIPTION
**Pull request by TyrantLucifer**
## Purpose of this pull request

1. Optimize code structure
2. Support genenrate `SeaTunnelRow`
3. Support user-defined map size
4. Support user-defined array size
5. Support user-defined string length
6. Support user-defined bytes length

## Check list

* [x] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [x] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
@laglangyue @EricJoy2048 @CalvinKirs @hailin0 Please review.

**Comment by laglangyue** [comment]
LGTM

File: seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeRandomData.java [hunk 1]
```diff
@@ -51,12 +39,14 @@
 public class FakeRandomData {
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeRandomData.java]:
```suggestion
public class FakeDataGenerator {
```

**Review by hailin0 — COMMENTED** [review]
LGTM

File: seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeRandomData.java (hunk 1 above)
Comment (TyrantLucifer) [reply seatunnel-connectors-v2/connector-fake/src/main/java/org/apache/seatunnel/connectors/seatunnel/fake/source/FakeRandomData.java]: Good advice.

**Review by EricJoy2048 — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**